Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py

```diff
@@ -6342,6 +6343,227 @@ def test_set_mirror_last_run_in_6_5(self, mocker):
             assert set_last_run.called is False
 
 
+class TestFetchWithLookBack:
+
+    LAST_RUN = {}
+    INCIDENTS = [
+        {
+            'incident_id': 1,
+            'created': (datetime.utcnow() - timedelta(hours=3)).strftime('%Y-%m-%dT%H:%M:%S')
+        },
+        {
+            'incident_id': 2,
+            'created': (datetime.utcnow() - timedelta(hours=1)).strftime('%Y-%m-%dT%H:%M:%S')
+        },
+        {
+            'incident_id': 3,
+            'created': (datetime.utcnow() - timedelta(minutes=29)).strftime('%Y-%m-%dT%H:%M:%S')
+        },
+        {
+            'incident_id': 4,
+            'created': (datetime.utcnow() - timedelta(minutes=19)).strftime('%Y-%m-%dT%H:%M:%S')
+        },
+        {
+            'incident_id': 5,
+            'created': (datetime.utcnow() - timedelta(minutes=9)).strftime('%Y-%m-%dT%H:%M:%S')
+        }
+    ]
+
+    NEW_INCIDENTS = [
+        {
+            'incident_id': 6,
+            'created': (datetime.utcnow() - timedelta(minutes=49)).strftime('%Y-%m-%dT%H:%M:%S')
+        },
+        {
+            'incident_id': 7,
+            'created': (datetime.utcnow() - timedelta(minutes=25)).strftime('%Y-%m-%dT%H:%M:%S')
+        },
+        {
+            'incident_id': 8,
+            'created': (datetime.utcnow() - timedelta(minutes=23)).strftime('%Y-%m-%dT%H:%M:%S')
+        }
+    ]
+    
+    def example_fetch_incidents(self):
+        """
+        An example fetch for testing
+        """
+
+        from CommonServerPython import get_fetch_run_time_range, filter_incidents_by_duplicates_and_limit, \
+            update_last_run_object
+
+        incidents = []
+
+        params = demisto.params()
+        fetch_limit_param = params.get('limit')
+        look_back = int(params.get('look_back', 0))
+        first_fetch = params.get('first_fetch')
+        time_zone = params.get('time_zone', 0)
+
+        last_run = demisto.getLastRun()
+        fetch_limit = last_run.get('limit') or fetch_limit_param
+
+        if not last_run.get('limit'):
+            last_run['limit'] = fetch_limit
+
+        start_fetch_time, end_fetch_time = get_fetch_run_time_range(last_run, first_fetch, look_back, time_zone)
+
+        query = self.build_query(start_fetch_time, end_fetch_time, fetch_limit)
+        incident_res = self.get_incidents_request(query)
+
+        incidents = filter_incidents_by_duplicates_and_limit(incident_res, last_run, fetch_limit_param, 'incident_id')
+
+        last_run = update_last_run_object(last_run, incidents, fetch_limit_param, start_fetch_time, end_fetch_time,
+                                          look_back, 'created', 'incident_id')
+
+        demisto.setLastRun(last_run)
```

Comment:
```suggestion
        new_last_run = update_last_run_object(last_run, incidents, fetch_limit_param, start_fetch_time, end_fetch_time,
                                          look_back, 'created', 'incident_id')
        last_run.update(new_last_run)
        demisto.setLastRun(last_run)
```
because last_run is not updated in place, need to make sure that other items in this object are not overridden.